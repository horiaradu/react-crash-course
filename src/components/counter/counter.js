import React from "react";
import PropTypes from "prop-types";
import { useState } from "react";

export const Counter = props => {
  const [count, setCount] = useState(0);

  const increment = () => setCount(count + 1);

  return (
    <p>
      <button onClick={increment}>click</button>
      <br />
      <br />
      <span>
        Counter ({props.name}): {count}
      </span>
    </p>
  );
};

Counter.propTypes = {
  name: PropTypes.string
};

Counter.defaultProps = {
  name: "fancy counter"
};
