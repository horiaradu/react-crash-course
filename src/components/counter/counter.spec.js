import React from "react";
import ReactDOM from "react-dom";
import { Counter } from "./counter";

describe("Counter", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Counter />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has a default name", () => {
    const div = document.createElement("div");
    const counter = ReactDOM.render(<Counter />, div);

    expect(counter.props.name).toEqual("fancy counter");
  });

  it("reads the name from props", () => {
    const div = document.createElement("div");
    const counter = ReactDOM.render(<Counter name="my counter" />, div);

    expect(counter.props.name).toEqual("my counter");
  });

  it("has a default value of 0", () => {
    const div = document.createElement("div");
    const counter = ReactDOM.render(<Counter />, div);

    expect(counter.state.count).toEqual(0);
  });

  it("increments the value", () => {
    const div = document.createElement("div");
    const counter = ReactDOM.render(<Counter />, div);

    counter.increment();
    expect(counter.state.count).toEqual(1);
  });

  it("matches the snapshot", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Counter />, div);
    expect(div).toMatchSnapshot();
  });
});
