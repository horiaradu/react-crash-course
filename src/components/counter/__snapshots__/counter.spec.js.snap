// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter matches the snapshot 1`] = `
<div>
  <p>
    <button>
      click
    </button>
    <br />
    <br />
    <span>
      Counter (
      fancy counter
      ): 
      0
    </span>
  </p>
</div>
`;
